e2e/nx-plugin-e2e/tests/configuration.e2e.test.ts

```diff
@@ -1,35 +1,138 @@
-import { expect } from 'vitest';
+import { Tree } from '@nx/devkit';
+import { rm } from 'node:fs/promises';
+import { join } from 'node:path';
+import { afterEach, expect } from 'vitest';
+import { generateCodePushupConfig } from '@code-pushup/nx-plugin';
+import {
+  generateWorkspaceAndProject,
+  materializeTree,
+} from '@code-pushup/test-nx-utils';
 import { removeColorCodes } from '@code-pushup/test-utils';
-import { executeProcess } from '@code-pushup/utils';
+import { distPluginPackage, executeGenerator } from '../mocks/utils';
+
+function executeConfigurationGenerator(
+  args: string[],
+  cwd: string = process.cwd(),
+) {
+  return executeGenerator(args, {
+    bin: distPluginPackage(cwd),
+    generator: 'configuration',
+    cwd,
+  });
+}
 
 describe('nx-plugin g configuration', () => {
-  it('should run configuration generator on react-todos-app', async () => {
-    const { code, stderr, stdout } = await executeProcess({
-      command: 'npx',
-      args: [
-        'nx',
-        'g',
-        '@code-pushup/nx-plugin:configuration react-todos-app --dryRun',
-      ],
-    });
+  let tree: Tree;
+  const project = 'my-lib';
+  const projectRoot = join('libs', project);
+  const baseDir = 'tmp/nx-plugin-e2e/generators/configuration';
+
+  const generatorExecMsgRegex = (cwd: string) =>
+    `NX  Generating ${distPluginPackage(cwd)}:configuration`;
+  const createConfigMsgRegex = /^CREATE.*code-pushup.config.ts/m;
+  const updateProjectMsgRegex = /^UPDATE.*project.json/m;
+
+  beforeEach(async () => {
+    tree = await generateWorkspaceAndProject(project);
+  });
+
+  afterEach(async () => {
+    await rm(baseDir, { recursive: true, force: true });
+  });
+
+  it('should inform about dry run', async () => {
+    const cwd = join(baseDir, 'configure');
+    await materializeTree(tree, cwd);
+
+    const { stderr } = await executeConfigurationGenerator(
+      [project, '--dryRun'],
+      cwd,
+    );
+
+    const cleanedStderr = removeColorCodes(stderr);
+    expect(cleanedStderr).toContain(
+      'NOTE: The "dryRun" flag means no changes were made.',
+    );
+  });
+
+  it('should generate conde-pushup.config.ts file and add target to project.json', async () => {
```

Comment:
I don't see how you're asserting that the target was added.

Reply:
As in mentioned here:
https://github.com/code-pushup/cli/pull/785#discussion_r1705579457